sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SpecificationLoader.java

```diff
@@ -34,18 +35,28 @@
 
 public class SpecificationLoader {
   private static final Logger LOG = Loggers.get(SpecificationLoader.class);
-  private final Map<String, Rule> rulesMappedToKey;
+
+  private static final Set<String> DEFAULT_SPECIFICATIONS = Set.of(
+    "alibaba.yaml",
+    "aws.yaml",
+    "azure.yaml",
+    "gcp.yaml",
+    "google-api.yaml",
+    "ibm.yaml",
+    "mws.yaml"
+  );
+  private final Map<String, List<Rule>> rulesMappedToKey;
 
   public SpecificationLoader() {
-    this("org/sonar/plugins/secrets/configuration/", Collections.emptySet());
+    this("org/sonar/plugins/secrets/configuration/", DEFAULT_SPECIFICATIONS);
   }
 
   public SpecificationLoader(String specificationLocation, Set<String> specifications) {
     rulesMappedToKey = initialize(specificationLocation, specifications);
   }
 
-  private static Map<String, Rule> initialize(String specificationLocation, Set<String> specifications) {
-    Map<String, Rule> keyToRule = new HashMap<>();
+  private static Map<String, List<Rule>> initialize(String specificationLocation, Set<String> specifications) {
+    Map<String, List<Rule>> keyToRule = new HashMap<>();
```

Comment:
Why there should be multiple rules to a single key? Same key but multiple rules for the different providers?

Reply:
@nils-werner-sonarsource 
Because of the restructuring of detection, it is now only possible to have a single pattern detection for a rule. So now matchEither for two patterns, only for auxiliaryPatterns.
Because of this and to use different messages for the same ruleKey, it is possible to have two rules in the specification which will raise an issue based on the same key